runtime/buffer.cc

```diff
@@ -28,6 +28,14 @@ std::size_t alloc_size(std::size_t rank, std::size_t elem_size, const dim* dims)
 
 std::size_t raw_buffer::size_bytes() const { return alloc_size(rank, elem_size, dims); }
 
+std::size_t raw_buffer::elem_count() const {
+  std::size_t result = 1;
+  for (std::size_t i = 0; i < rank; ++i) {
+    result *= std::max<index_t>(0, dims[i].extent());
```

Comment:
Wouldn't it return 0 if one of the extents is 0?

Reply:
It would, as intended, such a buffer is empty, isn't it? This function basically answers the question: "how many times will `for_each_element` call the callback for this buffer?" If any extent <= 0, the answer to that question is 0.

Reply:
I got confused by `max` there, I thought extent is always >=0.